### PR TITLE
Add section under Implementation on Command Validity Based on Application Mode

### PR DESCRIPTION
**Pull request by ruiyigan**
- Explain how the validity of user's command depends on Application Mode

### CODE REVIEW

File: docs/images/CommandValidityActivityDiagram.png
Comment (rockman007372) [inline docs/images/CommandValidityActivityDiagram.png:1]: I think the "Checking validitiy of command in current application" should not be inside the branch node. I think you can put it in an action node instead to stay consistent with the text book!

image

Something like this, but "Checking validitiy of command in current mode" replaces "check product"

**Review by rockman007372 — CHANGES_REQUESTED** [review]
LGTM but I have one minor issue with the UML sequence diagram!

File: docs/images/CommandValidityActivityDiagram.png
Comment (ruiyigan) [reply docs/images/CommandValidityActivityDiagram.png:1]: Good point! Made the changes. Can check if it is ok

**Review by rockman007372 — APPROVED** [review]
LGTMM

**Pull request merged**